sqlglot/dialects/duckdb.py

```diff
@@ -66,12 +66,11 @@ def _sort_array_reverse(args):
     return exp.SortArray(this=seq_get(args, 0), asc=exp.false())
 
 
-def _struct_pack_sql(self, expression):
+def _struct_sql(self, expression):
     args = [
-        self.binary(e, ":=") if isinstance(e, exp.EQ) else self.sql(e)
-        for e in expression.expressions
+        f"'{e.name or e.this.name}': {self.sql(e, 'expression')}" for e in expression.expressions
```

Comment:
The `e.name or e.this.name` part takes care of both the `struct_pack` call, which contains EQ expression, and the brace syntax, which uses Slice expressions.

Reply:
may be safer to do like

self.sql(exp.Literal.string(e.name)) to have escapes